File: Lambda/src/Lambda/Model/GameResponse.cs
using System.Diagnostics.CodeAnalysis;

namespace Lambda.Model;

public record GameResponse
{
    public GameResponse()
    {
    }

    [SetsRequiredMembers]
    public GameResponse(string response, string locationName, int moves, int score)
    {
        Response = response;
        LocationName = locationName;
        Moves = moves;
        Score = score;
    }

    public required string Response { get; init; }

    public required string LocationName { get; init; }

    public required int Moves { get; init; }

    public required int Score { get; init; }
}

File: shared-web-types/src/GameResponse.ts
export interface GameResponse {
    score: number;
    moves: number;
    time?: number;
    locationName: string;
    /**
     * The room's own identity - the location class's name. The display name is not one:
     * two rooms are called "Clearing", two "Cave", four "Forest".
     */
    locationKey?: string;
    response: string;
    inventory: string[];
    exits: string[];
    actionsAvailableFromInventory?: Record<string, string[]>;
    actionsAvailableFromLocation?: Record<string, string[]>;
    /**
     * True when the player is somewhere unlit. The server already withholds the room's
     * exits and action chips in that case (issue #238); the client withholds the room
     * artwork, which would otherwise show a picture of a room the player cannot see.
     */
    itIsDarkHere?: boolean;
}
